### ISSUE TITLE
Migrate to TF2.x API

### ISSUE THREAD

**Issue by remicres** [remicres/sr4rs#55]
I don't have the time right now to do that, but it should not be complicated. **External contributions are welcome to develop this feature!**

# Benefits

- Code easier to maintain
- Less code, simpler code thanks to OTBTF's [`ModelBase`](https://github.com/remicres/otbtf/blob/develop/otbtf/model.py) class (Built on top of `keras.model`)
- Training will benefit from keras framework, hence
  - Validation after each epoch,
  - Model saving strategies (e.g. save best model depending on one validation metric),
  - Training strategies (decay, early stopping, etc)
  - etc

# TODOs

- [ ] Refactor the model to work using TensorFlow V2 API and OTBTF: the new model must inherit from `ModelBase` from OTBTF
  - An example how to implement a FCNN model using OTBTF `ModelBase` [here](https://github.com/remicres/otbtf/tree/develop/otbtf/examples/tensorflow_v2x/fcnn)
  - An example how to implement GAN in `tf.keras.model` [here](https://keras.io/examples/generative/conditional_gan/#creating-a-conditionalgan-model)
